controllers/index/xjoinindexvalidator_iteration.go

```diff
@@ -164,19 +164,11 @@ func (i *XJoinIndexValidatorIteration) ParsePodResponse() (validation.Validation
 	podLogOpts := v1.PodLogOptions{}
 	req := i.ClientSet.CoreV1().Pods(i.Instance.GetNamespace()).GetLogs(i.ValidationPodName(), &podLogOpts)
 
-	podLogs, err := req.Stream(i.Context)
+	logString, err := i.PodLogReader.GetLogs(i.ValidationPodName(), i.Instance.GetNamespace())
 	if err != nil {
 		return response, errors.Wrap(err, 0)
 	}
-	defer podLogs.Close()
-
-	buf := new(bytes.Buffer)
-	_, err = io.Copy(buf, podLogs)
-	if err != nil {
-		return response, errors.Wrap(err, 0)
-	}
-	str := buf.String()
-	strArray := strings.Split(str, "\n")
+	strArray := strings.Split(logString, "\n")
```

Comment:
Do we want to let all logs to be fully read into memory when we're getting last lines out of the logs? Can we use Reader pattern and keep the memory utilization lower?

I'd recommend to create a function like `GetLastPodLogLines` (with a parameter on how many lines to return).

Reply:
thanks, I filed a task for this: https://issues.redhat.com/browse/ESSNTL-4454